google/auth/headful.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Headful Credentials.
+Headful Credentials are …
+
+Example headful credential:
+{
+  "type": "external_account_authorized_user",
+  "audience": "//iam.googleapis.com/locations/global/workforcePools/$WORKFORCE_POOL_ID/providers/$PROVIDER_ID",
+  "refresh_token": "refreshToken",
+  "token_url": "https://sts.googleapis.com/v1/oauth/token",
+  "token_info_url": "https://sts.googleapis.com/v1/instrospect"
+  "client_id": "clientId",
+  "client_secret": "clientSecret"
+}
+"""
+
+_HEADFUL_JSON_TYPE = "external_account_authorized_user"
+
+class Credentials(external_account.Credentials):
```

Comment:
I think this cred should be standalone and not extend from the existing external account creds

Reply:
Done

Reply:
@lsirac any reason not to extend from external account creds?

Reply:
There isn't much overlap between the creds. The external_account creds are for headless flows where the credential can be sourced through the method specified in the config file. Would add confusion if we overload them further.